build_tools/ci/cpu_comparison/run_test.py

```diff
@@ -509,8 +518,43 @@ def run_all(
     parser.add_argument(
         "vitis_dir", nargs="?", default="/opt/Xilinx/Vitis/2024.1", type=abs_path
     )
-    parser.add_argument("--return-on-fail", action="store_true", default=True)
-    parser.add_argument("-v", "--verbose", action="store_true", default=True)
+
+    # This (and other boolean flags) could be made more 'slick' by using
+    # `action='store_true'` in the `add_argument` call, but this has
+    # problems with the default value of 1. It could be also be made nicer
+    # by using type=bool, but this also has issues. So going with this
+    # clunky design for now (feel free to improve).
+    parser.add_argument(
+        "--return_on_fail",
+        nargs="?",
+        default=1,
+        type=int,
```

Comment:
what's the issue with bool and default=True?

Reply:
Trying to pass set it false at the command (=False, =0, ...) doesn't seem to do anything. No matter what I do it is True. 

With integers, I can pass it False / 0 at the command line and it actually makes it false.

Reply:
not sure - maybe something having to do with `add_argument` automatically injecting `action="store_true"` or something. anyway at least we should "cast"/"convert" to bool; see suggestion below.